### PR TITLE
Bump version of google-oauth-plugin and fix the tests

### PR DESCRIPTION
**Pull request by bzzitsme**
This PR introduces updates to the tests in response to the recent changes made in the google-oauth-plugin (PR [#201](https://github.com/jenkinsci/google-oauth-plugin/pull/201))
The key change involves the restructuring of `AbstractGoogleRobotCredentialsDescriptor`, which has now been integrated as a static inner class within `GoogleRobotCredentials`.

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by car-roll — APPROVED** [review]

File: src/test/java/com/google/jenkins/plugins/storage/util/GoogleRobotCredentialsDescriptor.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * In earlier versions, {@code AbstractGoogleRobotCredentialsDescriptor} was a standalone
+ * abstract class responsible for handling descriptors for Google robot account credential extensions.
+ * Now, {@code AbstractGoogleRobotCredentialsDescriptor} has been restructured and
+ * integrated as a static inner class within {@code GoogleRobotCredentials}.
+ *
+ * This change affects how the descriptor is accessed and used within the codebase. No major
+ * changes in the API usage are expected, but minor adjustments might be necessary in how the class
+ * is instantiated or accessed
+ */
+public class GoogleRobotCredentialsDescriptor extends GoogleRobotCredentials {
```
Comment (amuniz) [inline src/test/java/com/google/jenkins/plugins/storage/util/GoogleRobotCredentialsDescriptor.java]: Why are you copy&pasting this class here?
No tests in this plugin should depend on a protected inner class from another plugin.

Comment (bzzitsme) [reply src/test/java/com/google/jenkins/plugins/storage/util/GoogleRobotCredentialsDescriptor.java]: Tests in this plugin were using `AbstractGoogleRobotCredentialsDescriptor` but this descriptor class was [deleted](https://github.com/jenkinsci/google-oauth-plugin/pull/201/files#diff-4caf35987ce7b76e4d3e7c40fceffcf5658d1ac405be66a2a8feccebe52d68e7) from google-oauth-plugin and put in [GoogleRobotCredentials](https://github.com/jenkinsci/google-oauth-plugin/blob/develop/src/main/java/com/google/jenkins/plugins/credentials/oauth/GoogleRobotCredentials.java#L169)

Comment (amuniz) [reply src/test/java/com/google/jenkins/plugins/storage/util/GoogleRobotCredentialsDescriptor.java]: Yeah, but why do this tests need `GoogleRobotCredentialsDescriptor`?

Comment (bzzitsme) [reply src/test/java/com/google/jenkins/plugins/storage/util/GoogleRobotCredentialsDescriptor.java]: This [test](https://github.com/jenkinsci/google-storage-plugin/pull/277/commits/cb891b873eb5d9e622ad07b9f6dade4c471d3ead#diff-ff6efee22c1f22c7a1f1aafa5e15995fd91b9c871818229160fa4c85d807e21aR57) was failing for me, but it seems to be some local issue and now its working fine. 
I've [removed](https://github.com/jenkinsci/google-storage-plugin/pull/277/commits/cb891b873eb5d9e622ad07b9f6dade4c471d3ead) the descriptor test class, as its not necessary to have it in the tests

**Review by amuniz — APPROVED** [review]

**Review by Pldi23 — APPROVED** [review]

**Comment by bzzitsme** [comment]
@jtnord @basil could you please review/merge this PR? 🙂

**Review by jtnord — APPROVED** [review]

**Pull request merged**